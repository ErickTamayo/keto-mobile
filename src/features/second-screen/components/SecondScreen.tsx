import * as React from 'react'
import { Text, View, Button } from 'react-native'
import { NavigationInjectedProps } from 'react-navigation'
import st from '../../../styles'
import NavigationOptions from '../../../constants/NavigationOptions'

export default class SecondScreen extends React.Component<NavigationInjectedProps, object> {
  public static navigationOptions = NavigationOptions

  private title = (): string => 'Go Back'

  public render(): JSX.Element {
    const { navigation } = this.props

    return (
      <View style={[st.flex.f1, st.items.center, st.justify.center, st.bg.greyLightest]}>
        <Text>Second Screen</Text>
        <Button onPress={() => navigation.navigate('Start')} title={this.title()} color="#841584" />
      </View>
    )
  }
}
