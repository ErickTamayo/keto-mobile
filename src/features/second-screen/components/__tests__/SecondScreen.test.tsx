import 'react-native'
import * as React from 'react'
import { shallow } from 'enzyme'
import {
  NavigationScreenProp,
  NavigationRoute,
  NavigationParams,
  NavigationInjectedProps,
} from 'react-navigation'
import SecondScreen from '../SecondScreen'

describe('<SecondScreen />', () => {
  it('renders correctly', () => {
    const props: NavigationInjectedProps = {
      navigation: {} as NavigationScreenProp<NavigationRoute<NavigationParams>, NavigationParams>,
    }

    const wrapper = shallow(<SecondScreen {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
