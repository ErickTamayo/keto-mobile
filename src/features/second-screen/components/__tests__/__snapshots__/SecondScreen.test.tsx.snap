// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SecondScreen /> renders correctly 1`] = `
<Component
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#f7f7f7",
      },
    ]
  }
>
  <Component>
    Second Screen
  </Component>
  <Button
    color="#841584"
    onPress={[Function]}
    title="Go Back"
  />
</Component>
`;
